File: templates/hytale/basic-java/src/main/java/{{PACKAGE_PATH}}/{{MAIN_CLASS_NAME}}.java
package {{PACKAGE_NAME}};

import com.hypixel.hytale.server.core.plugin.JavaPlugin;
import com.hypixel.hytale.server.core.plugin.JavaPluginInit;
import com.hypixel.hytale.logger.HytaleLogger;

import java.util.logging.Level;

public class {{MAIN_CLASS_NAME}} extends JavaPlugin {

    private static final HytaleLogger LOGGER =
            HytaleLogger.forEnclosingClass();

    public {{MAIN_CLASS_NAME}}(JavaPluginInit init) {
        super(init);
    }

    @Override
    protected void setup() {
        LOGGER.at(Level.INFO).log("[Template] Setting up...");
        LOGGER.at(Level.INFO).log("[Template] Setup complete!");
    }

    @Override
    protected void start() {
        LOGGER.at(Level.INFO).log("[Template] Started!");
    }

    @Override
    protected void shutdown() {
        LOGGER.at(Level.INFO).log("[Template] Shutting down...");
    }
}

File: templates/hytale/basic-kotlin/src/main/kotlin/{{PACKAGE_PATH}}/{{MAIN_CLASS_NAME}}.kt
package {{PACKAGE_NAME}}

import com.hypixel.hytale.logger.HytaleLogger
import com.hypixel.hytale.server.core.plugin.JavaPlugin
import com.hypixel.hytale.server.core.plugin.JavaPluginInit
import java.util.logging.Level

class {{MAIN_CLASS_NAME}}(
init: JavaPluginInit
) : JavaPlugin(init) {

    private val logger =
        HytaleLogger.forEnclosingClass()

    override fun setup() {
        logger.at(Level.INFO).log("[{{PROJECT_NAME}}] Setting up...")
        logger.at(Level.INFO).log("[{{PROJECT_NAME}}] Setup complete!")
    }

    override fun start() {
        logger.at(Level.INFO).log("[{{PROJECT_NAME}}] Started!")
    }

    override fun shutdown() {
        logger.at(Level.INFO).log("[{{PROJECT_NAME}}] Shutting down...")
    }
}